code/modules/projectiles/updated_projectiles/guns/flamer.dm

```diff
@@ -352,6 +354,145 @@
 		if(!current_mag || !current_mag.current_rounds)
 			return
 
+/obj/item/weapon/gun/flamer/M240T/examine(mob/user)
+	..()
+	to_chat(user, "<span class='notice'>Its hydro cannon contains [get_water()]/[max_water] units of water!</span>")
+
+//Returns the amount of water in the M240T
+/obj/item/weapon/gun/flamer/M240T/proc/get_water()
+	return reagents.get_reagent_amount("water")
+
+/obj/item/weapon/gun/flamer/M240T/New()
+	..()
+	var/datum/reagents/R = new/datum/reagents(max_water)
+	reagents = R
+	R.my_atom = src
+	R.add_reagent("water", max_water)
+
+	var/obj/item/attachable/hydro_cannon/G = new(src)
+	G.icon_state = ""
+	G.Attach(src)
+	update_attachable(G.slot)
+	G.icon_state = initial(G.icon_state)
+
+/obj/item/weapon/gun/flamer/M240T/Fire(atom/target, mob/living/user, params, reflex)
+	if(active_attachable && istype(active_attachable, /obj/item/attachable/hydro_cannon))
+		hydrocannon(target,src,user) //Fire it.
+		last_fired = world.time
+		return
+	..()
+
+/obj/item/weapon/gun/flamer/M240T/proc/hydrocannon(atom/target, obj/item/weapon/gun/gun, mob/user)
+
+	if (src.reagents.total_volume < 1)
+		to_chat(user, "\red \The [src]'s hydro cannon is empty.")
+		return
+
+	if (world.time < src.last_use + 20)
+		return
+
+	user.visible_message("<span class='danger'>[user] fires the [src]'s hydro cannon!</span>", \
+	"<span class='warning'>You fire the [src]'s hydro cannon.</span>",)
+
+	last_use = world.time
+
+	playsound(user.loc, 'sound/effects/extinguish.ogg', 52, 1, 7)
+
+	var/direction = get_dir(user,target)
+
+	if(user.buckled && isobj(user.buckled) && !user.buckled.anchored )
```

Comment:
This code is copied from `/obj/item/tool/extinguisher/afterattack`. You might want to make it a proc as not to have repeated code, and to also implement it in the `afterattack`